### PR TITLE
207 implementation - OEE, EEF and OPS

### LINKED ISSUE #207: Implementation, OEE, EEF and OPS
Write about:

- [x] Error function
- [x] OEE
- [x] Optimal Production Speed
- [x] Add graphs and tables to the appendix

### PR DESCRIPTION
**Pull request by Kechr**

### CODE REVIEW

File: project_report/tex/implementation.tex [hunk 1]
```diff
@@ -86,3 +86,170 @@ \subsection{OPC-UA-Client}
 subscriptions will add data to the object. When done the Batch object will run
 through its logic and finally export itself as a JSON so it can be sent to the
 API. 
+
+
+\subsection{Overall Equipment Effectiveness}
+When a batch is complete, the calculation of the OEE takes place, and the data
+stored in the batch class is used for this.\\
```
Comment (Bokaiii) [inline project_report/tex/implementation.tex]: OEE takes place, which the data stored in the batch class is used for.

File: project_report/tex/implementation.tex [hunk 2]
```diff
@@ -86,3 +86,170 @@ \subsection{OPC-UA-Client}
...
+
+For this project, the planned production time can not be estimated properly, as
+too many variables are unknown (shift lengths, the time it takes to refill
+ingredients, the time it takes to do maintenance work etc.). Therefore, it has
+been chosen to use the time it takes to produce a batch. The calculation can be
```
Comment (Bokaiii) [inline project_report/tex/implementation.tex]: chosen = decided

maybe: "...decided to ONLY use the time it takes to produce a batch."

File: project_report/tex/implementation.tex [hunk 3]
```diff
@@ -86,3 +86,170 @@ \subsection{OPC-UA-Client}
...
+
+Where the amount to produce divided by the machine speed gives us the production
+time in minutes, as the machine speed is products per minute. This can be
+multiplied by 60 to give the production time in seconds. To find the ideal cycle
+time the maximum machine speed is needed. 1 divided by this multiplied by 60
+then gives the theoretical minimum time it takes to produce one unit.
+
+\[IdealCycleTime = \frac{1}{maxMachineSpeed}\cdot60\]\\
+
+When these two have been calculated, it is possible to calculate the OEE using
```
Comment (Bokaiii) [inline project_report/tex/implementation.tex]: Maybe remove "two": "When these have been calculated"

File: project_report/tex/implementation.tex [hunk 4]
```diff
@@ -86,3 +86,170 @@ \subsection{OPC-UA-Client}
...
+\[IdealCycleTime = \frac{1}{300}\cdot60=0.2\]
+
+\[OEE = \left(\frac{97\cdot0,2}{30}\right)\cdot100 = 64,67\%\]\\
+
+The OEE for this specific batch would be 64,67\%, which means that the
+production line can be optimised further if needed.
+
+\subsection{Error Function}
+The error function associated with each product type is to be estimated. For
+this several batches should be produced to have some data to make the
```
Comment (Bokaiii) [inline project_report/tex/implementation.tex]: Comma after "this"

"make" the calculations with = "do" the calculations

File: project_report/tex/implementation.tex [hunk 5]
```diff
@@ -86,3 +86,170 @@ \subsection{OPC-UA-Client}
...
+calculations with. The values used to estimate the error function is the amount
```
Comment (Bokaiii) [inline project_report/tex/implementation.tex]: Values = variables

File: project_report/tex/implementation.tex [hunk 6]
```diff
@@ -86,3 +86,170 @@ \subsection{OPC-UA-Client}
...
+     \hline
+     Alcohol Free & \(y = 3E-05x^3 - 0,0021x^2 + 0,4464x + 10\) & \(R^2 = 0,9935\) \\
+     \hline
+    \end{tabularx}
+    \caption{Estimated Error Functions}
+    \label{table:eef}
+\end{table}
+
+\subsection{Optimal Production Speed}
+To find the optimal production speed the same data as when estimating the error
```
Comment (Bokaiii) [inline project_report/tex/implementation.tex]: Comma after speed

File: project_report/tex/implementation.tex [hunk 7]
```diff
@@ -86,3 +86,170 @@ \subsection{OPC-UA-Client}
...
+function is used. As the error simulation in the machine is a function of the
+normalised machine speed and the selected product, the normalised machine speed
+must be calculated. This is done by taking 100 over the maximum machine speed.
+This multiplied by chosen machine speed will then give the percentage of the
```
Comment (Bokaiii) [inline project_report/tex/implementation.tex]: Comma after "this"
"a" after "by"

File: project_report/tex/implementation.tex [hunk 8]
```diff
@@ -86,3 +86,170 @@ \subsection{OPC-UA-Client}
...
+
+\begin{figure}[ht]
+	\centering
+	\includegraphics[width=1\linewidth]{images/ops/pilsner.png}
+	\caption{Optimal Production Speed for Pilsner}
+	\label{figure:ops_pilsner}
+\end{figure}
+
+This plot is generated using Microsoft Excel. When the OEE for each tested
+machine speed is plotted, a third degree polynomial trend line is made and an
```
Comment (Bokaiii) [inline project_report/tex/implementation.tex]: made = applied

File: project_report/tex/implementation.tex [hunk 9]
```diff
@@ -86,3 +86,170 @@ \subsection{OPC-UA-Client}
...
+
+With an \(R^2\) of \(0,9905\) which is used to see how close to the data points
+the polynomial is (the closer to 1 the more accurate).\\
+
+To find the maximum of the function, the derivative is calculated:
+
+\[y' = -\frac{9x^2-668x+1057}{10000}\]
+
+To find the maximum this equation should be solved for \(x\) when \(y=0\). This
+gives \(72,60463\), which is percentage of the maximum machine speed which is
```
Comment (Bokaiii) [inline project_report/tex/implementation.tex]: This gives an x-value of 72,60463, which is the optimal machine speed for the pilsner product type, measured in percentage.

File: project_report/tex/implementation.tex [hunk 10]
```diff
@@ -86,3 +86,170 @@ \subsection{OPC-UA-Client}
...
+the most optimal production speed for the pilsner product type. 
+
+To make sure that it is a local maximum at \(72,60463\), the second derivative
```
Comment (Bokaiii) [inline project_report/tex/implementation.tex]: it = the calculated x-value is a local maximum.....

File: project_report/tex/implementation.tex [hunk 11]
```diff
@@ -86,3 +86,170 @@ \subsection{OPC-UA-Client}
...
+is calculated:
+
+\[y'' = -\frac{18x-668}{10000}\]
+
+Then this equation is solved for \(y\) when \(x=72,60463\), which gives
```
Comment (Bokaiii) [inline project_report/tex/implementation.tex]: Comma after "then"

Which gives a value of

File: project_report/tex/implementation.tex [hunk 12]
```diff
@@ -86,3 +86,170 @@ \subsection{OPC-UA-Client}
...
+     \hline
+     Alcohol Free & 76,35\% of 125 & 95,44 products/minute \\
+     \hline
+    \end{tabularx}
+    \caption{Optimal Production Speed}
+    \label{table:ops}
+\end{table}
+
+
+The reason for the result of the stout is, that the function has no local maxima
```
Comment (Bokaiii) [inline project_report/tex/implementation.tex]: Never comma in front of "that"

File: project_report/tex/implementation.tex [hunk 13]
```diff
@@ -86,3 +86,170 @@ \subsection{OPC-UA-Client}
...
+in the range (0 to 200) but it keeps increasing, as seen in the graph for the
```
Comment (Bokaiii) [inline project_report/tex/implementation.tex]: Maybe split this sentence into two:

"... in the range (0 to 200). Instead, it keeps increasing, as seen..."

File: project_report/tex/theory_and_methods.tex [hunk 14]
```diff
@@ -4,15 +4,18 @@ \subsection{Theory}
 \subsubsection{Overall Equipment Effectiveness}
 The \textbf{O}verall \textbf{E}quipment \textbf{E}ffectiveness, OEE, is a
 measurement of how effective a given production line is. It identifies what
-percentage of the manufacturing time that is truly productive. The formula for
-the OEE can be seen in equation \ref{eq:2}
-\begin{equation} \label{eq:2}
-    OEE = \frac{Good Count * Ideal Cycle Time}{Planned Produciton Time}
+percentage of the manufacturing time that \myworries{is truly productive}. The formula for
```
Comment (Bokaiii) [inline project_report/tex/theory_and_methods.tex]: Maybe "truly productive" = "fully utilised"

**Review by Bokaiii — CHANGES_REQUESTED** [review]

**Review by Bokaiii — APPROVED** [review]
Gut Job

**Review by Joergenson — APPROVED** [review]

**Pull request merged**